frontend/src/app/modules/lessons/lesson/lesson.component.ts

```diff
@@ -71,33 +75,81 @@ export class LessonComponent extends BaseComponent implements OnInit {
         });
     }
 
-    enrollLesson() {
-        this.userService.enrollUserToLesson(this.lesson.id)
-            .pipe(this.untilThis)
-            .subscribe({ next: (lesson) => {
-                this.lesson.isSubscribed = true;
-                this.lesson.subscribersCount = lesson.subscribersCount;
-                this.notificationService.showSuccess(`You successfully registered for lesson ${this.lesson.title}`, 'Success!');
-            },
-            error: () => this.notificationService.showError(`Failed to register for lesson ${this.lesson.title}`, 'Failed!') });
+    buttonAction() {
+        if (this.page === 'timetable') {
+            this.userService
+                .enrollUserToLesson(this.lesson.id)
+                .pipe(this.untilThis)
+                .subscribe({
+                    next: (lesson) => {
+                        this.lesson.isSubscribed = true;
+                        this.lesson.subscribersCount = lesson.subscribersCount;
+                        this.notificationService.showSuccess(
+                            `You successfully registered for lesson ${this.lesson.name}`,
+                            'Success!',
+                        );
+                    },
+                    error: () =>
+                        this.notificationService.showError(
+                            `Failed to register for lesson ${this.lesson.name}`,
+                            'Failed!',
+                        ),
+                });
+        }
+        if (this.page === 'teacher') {
+            this.lessonsService.cancelLesson(this.lesson.id, this.lesson).subscribe((data) => {
+                this.lesson.isCanceled = data.isCanceled;
+            });
+        }
     }
 
     ngOnInit(): void {
-        this.previewImage = this.lesson.videoId ? `//img.youtube.com/vi/${this.lesson.videoId}/maxresdefault.jpg` : '';
+        this.previewImage = this.lesson.youtubeVideoId
+            ? `//img.youtube.com/vi/${this.lesson.youtubeVideoId}/maxresdefault.jpg`
+            : '';
     }
 
     isDisabled() {
-        return this.lesson.isSubscribed || new Date() > this.lesson.startAt;
+        if (this.page === 'timetable') {
+            return this.lesson.isSubscribed || new Date() > new Date(this.lesson.startAt);
+        }
+        if (this.page === 'teacher') {
+            return this.lesson.isCanceled || new Date() > new Date(this.lesson.startAt);
+        }
+
+        return false;
```

Comment:
Create flag if you need to then, like 'isTeacherPage' or type in enum, don't use strings for it)